packages/server/src/workers/cron.ts

```diff
@@ -112,51 +111,52 @@ export async function addCronJobs(resource: Resource): Promise<void> {
     return;
   }
 
-  let cron;
-  // Validate the cron format
-  if (bot.cronTiming) {
-    cron = convertTimingToCron(bot.cronTiming);
-    if (!cron) {
-      logger.debug('cronTiming had the wrong format for a timed cron job');
-      return;
-    }
-  } else if (bot.cronString && isValidCron(bot.cronString)) {
-    cron = bot.cronString;
-  } else if (bot.cronString === '') {
-    await removeBullMQJobByKey(bot.id as string);
-    logger.debug(`no job for bot: ${bot.id}`);
-    return;
-  } else {
-    logger.debug('cronString had the wrong format for a timed cron job');
+  const oldCronStr = getCronStringForBot(previousVersion as Bot);
+  const newCronStr = getCronStringForBot(bot);
+  logger.info('Cron job for bot', { botId: bot.id, oldCronStr, newCronStr });
+
+  if (oldCronStr === newCronStr) {
+    // No change in cron job
     return;
   }
 
-  const cronObject = { repeat: { pattern: cron } };
+  if (previousVersion) {
+    logger.info('Removing cron job for bot', { botId: bot.id });
+    await queue.removeJobScheduler(bot.id as string);
+  }
 
-  // JobId and repeatable instructions
-  const jobOptions = { ...cronObject, jobId: bot.id };
-  await addCronJobData(
-    {
-      resourceType: bot.resourceType,
-      botId: bot.id as string,
-    },
-    jobOptions
-  );
+  if (newCronStr) {
+    logger.info('Adding cron job for bot', { botId: bot.id });
+    if (queue) {
+      await queue.add(
```

Comment:
Should we be using `upsertJobScheduler`?
https://docs.bullmq.io/guide/job-schedulers

Reply:
Oo, nice find. Yes, will update.  Thanks!